### PR TITLE
Fix some shadow root tests

### PR DESCRIPTION
**Pull request by sarvaje**
This PR fix some remaining issues in the tests for the shadow root commands:
- Get Element Shadow Root
- Find Element From Shadow Root
- Find Elements From Shadow Root

### CODE REVIEW

**Comment by sarvaje** [comment]
Hi, this is my first contribution and I can find why the tasks are failing. Can you help me, please?

**Comment by mathiasbynens** [comment]
(For context, this patch is blocking https://chromium-review.googlesource.com/c/chromium/src/+/3033190.)

**Comment by foolip** [comment]
@sarvaje what you've run into is https://github.com/web-platform-tests/wpt/issues/7660. Due to the tests that have been modified, probably too many other tests are considered affected and run 10 times. But there's not enough time to run the 10 times, so both of the task logs end with "Task timeout after 7200 seconds. Force killing container.":
https://community-tc.services.mozilla.com/tasks/McbMml-mQByDrxWWwHiE8w/runs/0/logs/public/logs/live.log
https://community-tc.services.mozilla.com/tasks/L5eda_kSTOKaDau2omgoCQ/runs/0/logs/public/logs/live.log

When this happens, the only recourse is to ping @web-platform-tests/admins asking someone to admin merge, basically ignoring the failed CI checks.

If the tests are in order and you have a review but can't merge it, feel free to ping me or @past for merging.

**Comment by sarvaje** [comment]
@foolip Thanks for your answer, before pinging @web-platform-tests/admins for the merge, should I wait for someone to approve my changes, right?

**Comment by foolip** [comment]
Right, you'll need someone to review the actual changes here. In all honestly that probably won't happen by just waiting, so you might want to look at who has recently written or reviewed tests close to these and ping them directly until you find a reviewer.

**Comment by sarvaje** [comment]
Ok, thanks!!

**Comment by sarvaje** [comment]
@whimboo I see you reviewed the PR that added the Shadow root tests. Can you take a look to my PR too? Thanks!

File: webdriver/tests/get_element_shadow_root/get.py
Comment (whimboo) [inline webdriver/tests/get_element_shadow_root/get.py:22]: The error type as used before was correct given that the window handle has been changed in line 20. Only for the same handle the `stale element` would be raised.

**Review by whimboo — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution! It's great to see these obvious mistakes to be fixed soon. Nevertheless I added some inline comments that have to be looked at.

File: webdriver/tests/find_element_from_shadow_root/find.py
Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: I was testing and `tag name` and `xpath` are not supported in a shadow root element. If you want to, I can add some test to see that an invalid argument is received for those values.

File: tools/webdriver/webdriver/client.py
Comment (sarvaje) [reply tools/webdriver/webdriver/client.py:381]: I was taking `class Element` as reference and it has `id` first. I will restore this change and apply the fix where the class is used.

File: webdriver/tests/find_element_from_shadow_root/find.py
Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: For testing I mean testing in `chromium` and `firefox` and they don't support them, but maybe the is just because they don't have them implemented?

Comment (whimboo) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: How exactly did you test? Note that at least Firefox doesn't have a WebDriver implementation for shadow root yet. It's being worked on https://bugzilla.mozilla.org/show_bug.cgi?id=1700073.

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: I was testing javascript. I'm taking also a look to the w3c https://www.w3.org/TR/dom41/ and looks like shadow root doesn't have a method to get by the tag name.

Comment (plehegar) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: you should look at https://dom.spec.whatwg.org/ (sorry, /TR/dom41/ is [updated to redirect](https://github.com/w3c/whatwg-coord/issues/5#issuecomment-895391219)).

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: Thanks @plehegar for the link. I still can't find anything about supporting getting an element by their tag name in a shadow root

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: Maybe I'm interpretating this wrong but, I'm associating the 'tag name' with the method `getElementsByTagName`.

File: tools/webdriver/webdriver/client.py
Comment (whimboo) [reply tools/webdriver/webdriver/client.py]: Feel free to also fix `Element.__init__()` so that it no longer causes confusion.

File: webdriver/tests/find_element_from_shadow_root/find.py
Comment (whimboo) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: I actually asked some developers and it turned out that `getElementsByTagName` returns `HTMLCollection`, which is a legacy thing. So it's probably the reason why it is not supported. Same seems to also apply to `XPath`.

I would propose that you leave in these two selectors for now, and file a new issue for discussion. We may have to remove both selector types for shadow root elements.

Comment (whimboo) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: Actually `xpath` we might be able to keep given that `XPathEvaluator` could always be created, and the `shadowRoot`  can be passed in as start node. That's because `ShadowRoot` is still a `DocumentFragment`.

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: > Actually xpath we might be able to keep given that XPathEvaluator could always be created, and the shadowRoot can be passed in as start node. That's because ShadowRoot is still a DocumentFragment.

Do you have an example? When I test them in the F12 tool (Chromium, Firefox) I get an error. 
In chromium it says `Uncaught DOMException: Failed to execute 'evaluate' on 'XPathExpression': The node provided is '#document-fragment', which is not a valid context node type.`
In firefox it says `Uncaught DOMException: Operation is not supported`

The code I have used is:
```js
const evalutaor = new XPathEvaluator();
const expression = evalutaor.createExpression('//input');
expression.evaluate(shadowroot)
```

> I would propose that you leave in these two selectors for now, and file a new issue for discussion. We may have to remove both selector types for shadow root elements.

I like the idea, so we can unblock this PR. I will update the PR and create the issue for further discussion.

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py:77]: I have restored this code and [open an issue](https://github.com/web-platform-tests/wpt/issues/30010) to follow up the discussion :)

File: tools/webdriver/webdriver/client.py
Comment (sarvaje) [reply tools/webdriver/webdriver/client.py]: Done :)

File: webdriver/tests/find_element_from_shadow_root/find.py
Comment (whimboo) [reply webdriver/tests/find_element_from_shadow_root/find.py]: As already mentioned on the newly filed issue please file an issue for webdriver spec. Then as best copy and paste the above conversation as reference. Thanks!

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py]: @whimboo the issue in the w3c repo is created and I have added a link to this conversation in the description of the issue.

Comment (sarvaje) [reply webdriver/tests/find_element_from_shadow_root/find.py]: The link to the w3c issue is: https://github.com/w3c/webdriver/issues/1610

**Review by whimboo — APPROVED** [review]
Looks fine to me now.

**Comment by sarvaje** [comment]
Thanks @whimboo.
The checks failing are because of a timeout. @foolip mention that I can ping @web-platform-tests/admins to bypass the checks and get the PR merged. How can I ping them, github doesn't recognize @web-platform-tests/admins as a know group or anything hehehe.

**Review by whimboo — CHANGES_REQUESTED** [review]
Actually tests fail because of the changes to the `Element` class:

> url = '/session/84b28476-db5d-488a-8bb4-da75dde1ec5f/element/<Session 84b28476-db5d-488a-8bb4-da75dde1ec5f>/value'

As such please revert these, to not cause massive extra work.

**Comment by sarvaje** [comment]
> As such please revert these, to not cause massive extra work.

done

**Review by whimboo — APPROVED** [review]
Yes, all tests passing now and stability ones timeout as expected. Thanks!

**Pull request merged**
